src/charm.py

```diff
@@ -307,6 +313,62 @@ def check_pgb_running(self):
 
         return True
 
+    def get_hostname_by_unit(self, unit_name: str) -> str:
+        """Create a DNS name for a PgBouncer unit.
+
+        Args:
+            unit_name: the juju unit name, e.g. "postgre-sql/1".
+
+        Returns:
+            A string representing the hostname of the PostgreSQL unit.
+        """
+        unit_id = unit_name.split("/")[1]
+        return f"{self.app.name}-{unit_id}.{self.app.name}-endpoints"
```

Comment:
There's a `relations.peers._get_unit_hostname()` already in the charm, but it uses units objects instead of unit names